packages/consumption/src/modules/attributes/AttributesController.ts

```diff
@@ -1288,4 +1288,25 @@ export class AttributesController extends ConsumptionBaseController {
 
         return ownSharedAttributeSuccessors;
     }
+
+    public async getRelationshipAttributesOfValueTypeToPeerWithGivenKeyAndOwner(key: string, owner: CoreAddress, valueType: string, peer: CoreAddress): Promise<LocalAttribute[]> {
+        const queryForRelationshipAttributesWithSameKey = {
+            "content.@type": "RelationshipAttribute",
+            "content.owner": owner.toString(),
+            "content.key": key,
+            "content.value.@type": valueType,
+            "shareInfo.peer": peer.toString(),
+            "shareInfo.thirdPartyAddress": { $exists: false },
+            "deletionInfo.deletionStatus": {
+                $nin: [
+                    LocalAttributeDeletionStatus.ToBeDeleted,
```

Comment:
I think it is debatable whether to allow a RelationshipAttribute that is in deletion to exist with the same `key`. As I can imagine cases in which it would be restrictive to prohibit this, I have allowed it for the time being. For example, if a RelationshipAttribute is being deleted because it is about to expire, but a new RelationshipAttribute with the same `key` is to be created, which is then valid for a longer period of time. It would feel unnatural to then have to wait for the final deletion of the former RelationshipAttribute. However, as I said, I think it is debatable. :D

I would say that `"DeletedByPeer"` RelationshipAttributes can be described as relics that embody history, but perhaps also still have a use. I don't know to what extent such RelationshipAttributes are ignored in the App, but that we often have to consider such RelationshipAttributes as edge cases in the Runtime, because it is no longer allowed to do so many things with them.

Reply:
Would it in your example be more appropriate to have the new Attribute succeed the toBeDeleted one rather than them coexisting? I also just saw that we have the option to answer the request with an attribute succession only for the Read- and Propose-, but not the CreateAttributeRequests. Is there a reason for that? I think this could allow requests with the same key if it can be accepted as an attribute succession.

Reply:
Probably, even if the succession can currently only be performed by the `owner` of the RelationshipAttribute. Requesting the succession of an existing RelationshipAttribute from its `owner` is a feature that we currently don't have.

Answering the Request with an Attribute Succession is only possible with IdentityAttributes and ThirdPartyRelationshipAttributes. For RelationshipAttributes, with all three RequestItems Read/Create/Propose, always a new RelationshipAttribute is created that cannot be part of an already existing succession chain. That was therefore not an option.